From: Michael Davidsaver <mdavidsaver@gmail.com>
Date: Thu, 8 Dec 2016 16:10:38 -0500
Subject: makeDbAndEdl.py add destination dir arguments

writing to ../.. makes assumptions about
where this script is run from which
aren't true when packaged.
---
 aravisGigEApp/src/makeDbAndEdl.py | 35 ++++++++++++++++++++++++-----------
 1 file changed, 24 insertions(+), 11 deletions(-)

diff --git a/aravisGigEApp/src/makeDbAndEdl.py b/aravisGigEApp/src/makeDbAndEdl.py
index 3afc3db..b587f16 100755
--- a/aravisGigEApp/src/makeDbAndEdl.py
+++ b/aravisGigEApp/src/makeDbAndEdl.py
@@ -1,10 +1,16 @@
-#!/bin/env python
+#!/usr/bin/env python
 import os, sys
 from xml.dom.minidom import parseString
-from optparse import OptionParser
 
-# parse args
-parser = OptionParser("""%prog <genicam_xml> <camera_name>
+def getargs():
+    from optparse import OptionParser
+
+    prefix = os.path.abspath(os.path.join(os.path.dirname(__file__),".."))
+    dbdir = os.path.join(prefix, "Db")
+    edldir = os.path.join(prefix, "opi", "edl")
+
+    # parse args
+    parser = OptionParser("""%prog <genicam_xml> <camera_name>
 
 This script parses a genicam xml file and creates a database template and edm 
 screen to go with it. The edm screen should be used as indication of what
@@ -14,9 +20,16 @@ a more sensible summary. The Db file will be generated in:
 and the edm files will be called:
   ../opi/edl/<camera_name>.edl
   ../opi/edl/<camera_name>-features.edl""")
-options, args = parser.parse_args()
-if len(args) != 2:
-    parser.error("Incorrect number of arguments")
+
+    parser.add_option("-D", "--db", help="write .db to this directory", default=dbdir)
+    parser.add_option("-E", "--edl", help="write .edl to this directory", default=edldir)
+
+    options, args = parser.parse_args()
+    if len(args) != 2:
+        parser.error("Incorrect number of arguments")
+    return options, args
+
+options, args = getargs()
 
 # Check the first two lines of the feature xml file to see if arv-tool left
 # the camera id there, thus creating an unparsable file
@@ -33,10 +46,10 @@ except:
 # parse xml file to dom object
 xml_root = parseString("".join(genicam_lines[start_line:]).lstrip())
 camera_name = args[1]
-prefix = os.path.abspath(os.path.join(os.path.dirname(__file__),".."))
-db_filename = os.path.join(prefix, "Db", camera_name + ".template")
-edl_filename = os.path.join(prefix, "opi", "edl", camera_name + ".edl")
-edl_more_filename = os.path.join(prefix, "opi", "edl", camera_name + "-features.edl")
+
+db_filename = os.path.join(options.db, camera_name + ".template")
+edl_filename = os.path.join(options.edl, camera_name + ".edl")
+edl_more_filename = os.path.join(options.edl, camera_name + "-features.edl")
 
 # function to read element children of a node
 def elements(node):
